unittests/cpp/gRPC/tests/ServiceCredentials_test.cpp

```diff
@@ -0,0 +1,312 @@
+}
+
+/*
+ * TEST 2 - ServiceCredentials with missing Authorization metadata.
+ */
+TEST_F(gRPCServiceCredentialsTests, InvalidCredentialsAuthz) {
+    grpc::AuthMetadataProcessor::OutputMetadata consumed, response; 
+
+    grpc::Status status = processor_.Process(input_metadata_, nullptr, &consumed, &response);
+    
+    // Setting expectations
+    EXPECT_EQ(status.error_code(), grpc::StatusCode::PERMISSION_DENIED);
+    EXPECT_EQ(status.error_message(), "No bearer token provided");
+}
+
+/*
+ * TEST 3 - ServiceCredentials with invalid token in Authorization metadata.
+ */
+TEST_F(gRPCServiceCredentialsTests, InvalidTokenInAuthzMetadata) {
+    grpc::AuthMetadataProcessor::OutputMetadata consumed, response;
+    input_metadata_.emplace("authorization", "Bearer not_a_valid_jwt_token");
+
+    grpc::Status status = processor_.Process(input_metadata_, nullptr, &consumed, &response);
+    
+    // Setting expectations
+    EXPECT_EQ(status.error_code(), grpc::StatusCode::PERMISSION_DENIED);
+    EXPECT_EQ(status.error_message(), "Invalid bearer token");
+}
+
+/*
+ * TEST 4 - Test with a string containing environment variables (e.g. ${HOME}) and verify replacement.
+ */
+TEST(gRPCExpandEnvVariablesTest, ReplaceExistingEnvVar) {
+    setenv("HOME", "/tmp/home", 1);
+    std::string input = "Path: ${HOME}/data";
+    catena::gRPC::expandEnvVariables(input);
+    
+    // Setting expectations
+    EXPECT_EQ(input, "Path: /tmp/home/data");
+    unsetenv("HOME");
+}
+
+/*
+ * TEST 5 - Test with a string containing an env variable that doesn't exist.
+ */
+TEST(gRPCExpandEnvVariablesTest, ReplaceNonExistingEnvVar) {
+    unsetenv("DOES_NOT_EXIST");
+    std::string input = "Path: ${DOES_NOT_EXIST}/data";
+    catena::gRPC::expandEnvVariables(input);
+    
+    // Setting expectations
+    EXPECT_EQ(input, "Path: /data");
+}
+
+/*
+ * TEST 6 - Test with a string containing no env variables.
+ */
+TEST(gRPCExpandEnvVariablesTest, NoEnvVarInString) {
+    std::string input = "Path: /static/data";
+    catena::gRPC::expandEnvVariables(input);
+    
+    // Setting expectations
+    EXPECT_EQ(input, "Path: /static/data");
+}
+
+/*
+ * TEST 7 - Test with FLAGS_secure_comms set to off
+ */
+TEST_F(gRPCServiceCredentialsTests, FlagsSecureCommsOff) {
+    absl::SetFlag(&FLAGS_secure_comms, "off");
+
+    // Setting expectations
+    auto creds = catena::gRPC::getServerCredentials();
+    auto insecure = grpc::InsecureServerCredentials();
+    EXPECT_EQ(typeid(*creds), typeid(*insecure));
+}
+
+/*
+ * TEST 8 - Test with FLAGS_secure_comms set to tls and mutual_authc set to false.
+ */
+TEST_F(gRPCServiceCredentialsTests, FLAGS_mutual_authcFalse) {
+    absl::SetFlag(&FLAGS_secure_comms, "tls");
+    absl::SetFlag(&FLAGS_mutual_authc, false);
+
+    // Setting expectations
+    auto creds = catena::gRPC::getServerCredentials();
+    auto insecure = grpc::InsecureServerCredentials();
+    EXPECT_NE(typeid(*creds), typeid(*insecure));
+    EXPECT_NE(creds, nullptr);
```

Comment:
```suggestion
    EXPECT_NE(creds, nullptr);
    EXPECT_NE(typeid(*creds), typeid(*insecure));
```
the null check should go first. typeid(*null) is undefined behaviour, so checking against null first makes for better test failure messages.